src/test/java/com/mathworks/ci/systemtests/TestResultVisualizationIT.java

```diff
@@ -0,0 +1,340 @@
+        assertTrue(notRunButtonText.contains("0"));
+        assertTrue(notRunButtonText.contains("Not Run"));
+
+        // Verify the test filenames shown
+        String[] expectedTestFiles = {" testMultiply ", " squareTest ", " testSquare ", " testSum "};
+        String[] testFiles = getTestFilesInTable(build);
+        assertEquals(testFiles, expectedTestFiles);
+
+        // Verify Diagnostics are shown for a failed test.
+        List<String[]> testsContentForAFile = getTestsInfoForATestFile("testMultiply", build);
+        assertEquals(testsContentForAFile.size(),1);
+        assertEquals(testsContentForAFile.get(0)[0].trim(), "testMultiplication"); // 'testMultiply has 'testMultiplication' test
+        assertNotNull(testsContentForAFile.get(0)[1]); // Diagnostics cant be null as 'testMultiply' fails
+        assertTrue(testsContentForAFile.get(0)[1].contains("testMultiply/testMultiplication"));
+        assertNotNull(testsContentForAFile.get(0)[2]); // Duration cannot be null
+
+        // Verify Diagnostics is empty for passed test
+        List<String[]> testsContent= getTestsInfoForATestFile("testSum", build);
+        assertEquals(testsContent.size(),1);
+        assertEquals(testsContent.get(0)[0].trim(), "testAddition"); // 'testSum has 'testAddition' test
+        assertEquals(testsContent.get(0)[1], ""); // No diagnostics as 'testAddition' passes
+    }
+
+    // Verify in matrix project
+    @Test
+    public void verifyTestResultsSummaryInMatrixProject() throws Exception {
+        String matlabRoot = System.getenv("MATLAB_ROOT");
+        String matlabRoot22b = System.getenv("MATLAB_ROOT_22b");
+        Assume.assumeTrue("Not running tests as MATLAB_ROOT_22b environment variable is not defined", matlabRoot22b != null && !matlabRoot22b.isEmpty());
+
+        Utilities.setMatlabInstallation("MATLAB_PATH_1", matlabRoot, jenkins);
+        Utilities.setMatlabInstallation("MATLAB_PATH_22b", matlabRoot22b, jenkins);
+
+        MatrixProject matrixProject = jenkins.createProject(MatrixProject.class);
+        MatlabInstallationAxis MATLABAxis = new MatlabInstallationAxis(Arrays.asList("MATLAB_PATH_1", "MATLAB_PATH_22b"));
+        matrixProject.setAxes(new AxisList(MATLABAxis));
+        matrixProject.setScm(new ExtractResourceSCM(Utilities.getURLForTestData()));
+
+        // Run tests through Run Build step
+        RunMatlabBuildBuilder buildtoolBuilder = new RunMatlabBuildBuilder();
+        buildtoolBuilder.setTasks("test");
+        matrixProject.getBuildersList().add(buildtoolBuilder);
+
+        MatrixBuild build = matrixProject.scheduleBuild2(0).get();
+
+        Combination c = new Combination(new AxisList(new MatlabInstallationAxis(Arrays.asList("MATLAB_PATH_1"))), "MATLAB_PATH_1");
+        MatrixRun run = build.getRun(c);
+        String[] buildResultSummaries = getTestResultSummaryFromBuildStatusPage(run);
+        List.of(buildResultSummaries).forEach(summary -> {
```

Comment:
`testResultSummaries`/`firstTestResultSummaries` in place of `buildResultSummaries`